File: Array_largeEntity.cpp
#include <iostream>
using namespace std; 

//This is telling me that in C++ large entities are passed by reference

void func(int arr[]){
   for(int i = 0; i<3; i++){
      arr[i] += 1;  
      cout << "In func: " << "arr["<<i<<"] is " << arr[i] << endl;
   }
}

int main(){
   int arr[] = {0,1,2};
   
   for(int i = 0; i<3; i++){
      cout << "Before Func: " << "arr["<<i<<"] is " << arr[i] << endl;
   }
   
   func(arr); 

   for(int i = 0; i<3; i++){
      cout << "After Func: " << "arr["<<i<<"] is " << arr[i] << endl;
   }

return 0;
}

File: Array_largeEntity.java
class Arrays {
   public static void main(String[] args) {
      int[] arr = {0,1,2};   
      System.out.println("Before Func: arr[0]: " + arr[0]);
      System.out.println("Before Func: arr[1]: " + arr[1]);
      System.out.println("Before Func: arr[2]: " + arr[2]);
      System.out.println("\n");
      func(arr);
      System.out.println("Before Func: arr[0]: " + arr[0]);
      System.out.println("Before Func: arr[1]: " + arr[1]);
      System.out.println("Before Func: arr[2]: " + arr[2]);
      System.out.println("\n");  
   }  
   public static void func(int[] a){
      a[0] += 1;
      a[1] += 1;
      a[2] += 1;
      
      System.out.println("Before Func: arr[0]: " + a[0]);
      System.out.println("Before Func: arr[1]: " + a[1]);
      System.out.println("Before Func: arr[2]: " + a[2]);
      System.out.println("\n");
   }
} 
